### ISSUE TITLE
Yeelight Lightstrip failing to connect.

### ISSUE THREAD

**Issue by mamsoudi** [rgb-master-team/RGBMaster#36]
Hey there,
I recently bought a Yeelight Lightstrip 1s, and I found out about RGBMaster and wanted to sync my devices, but weirdly enough while, it discovers the devices while searching whenever I try to check the device checkbox; it throws an error as shown below, I've tried resetting the devices itself, router, switching networks, etc. It just doesn't work.

![image](https://user-images.githubusercontent.com/8385009/104414940-09146700-55ac-11eb-8b2d-917ea1e57364.png)

I'd appreciate any help or suggestion, really looking forward to having my Mouse and the LED synced :(

**Comment by deankevorkian** [comment]
Hey! Actually we have been suffering some issues with the 1s, but mostly on led bulbs and not led strips. (Meaning we had people that were experiencing issues on Yeelight LED Bulb 1S) but not on the LED Light Strip 1S.

So a couple of questions - 
1. When opening the app, did it prompt you for giving network access for it? Did you allow it? Sometimes you can deny and have them discovered, but connections to it won't work.
2. Does the message appear after 5 seconds? (Meaning there was probably a timeout) or does it appear immediately?
3. Could you send relevant logs you find? You can grab them on C:\RGBMaster\Logs

And one thing that isn't a question but rather something to point out - 
There's a new pre-pre-pre-pre-super-pre-alpha version on the releases page - can you let me know if you have connectivity issues on that version too? it's not stable, but it contains some fixes and refactors we did for connectivity around Yeelights, and I wanna see if it has anything to do with it.

**Comment by mamsoudi** [comment]
Hey there, thanks for the quick response, and sorry for my delay,

About the questions,
1. I don't quite remember, I think I wasn't asked for Network Access; otherwise, I would've accepted 🤔 
2. Yes, so as soon I check the checkbox for the device, it goes into loading (or connecting mode), and the checkbox goes into the loading mode, but after a couple of seconds it just shows the error message above.
3. Sure thing, I actually managed to connect to it with Razer Chroma Studio, so I know it's not the device. Absolutely, I'll give it a shot tonight and attach the logs and everything for you with the alpha version, if I didn't work.

**Comment by Darkwolf1515** [comment]
> Hey there, thanks for the quick response, and sorry for my delay,
> 
> About the questions,
> 
> 1. I don't quite remember, I think I wasn't asked for Network Access; otherwise, I would've accepted 🤔
> 2. Yes, so as soon I check the checkbox for the device, it goes into loading (or connecting mode), and the checkbox goes into the loading mode, but after a couple of seconds it just shows the error message above.
> 3. Sure thing, I actually managed to connect to it with Razer Chroma Studio, so I know it's not the device. Absolutely, I'll give it a shot tonight and attach the logs and everything for you with the alpha version, if I didn't work.

Are you using 1.4 or 1.5? Cause 1.4 didn’t prompt me for network, but 1.5 did

**Comment by deankevorkian** [comment]
> Hey there, thanks for the quick response, and sorry for my delay,
> 
> About the questions,
> 1. I don't quite remember, I think I wasn't asked for Network Access; otherwise, I would've accepted 🤔 
> 2. Yes, so as soon I check the checkbox for the device, it goes into loading (or connecting mode), and the checkbox goes into the loading mode, but after a couple of seconds it just shows the error message above.
> 3. Sure thing, I actually managed to connect to it with Razer Chroma Studio, so I know it's not the device. Absolutely, I'll give it a shot tonight and attach the logs and everything for you with the alpha version, if I didn't work.

@mamsoudi did you give a shot to the 2.5 version?

**Comment by mamsoudi** [comment]
> > Hey there, thanks for the quick response, and sorry for my delay,
> > About the questions,
> > 
> > 1. I don't quite remember, I think I wasn't asked for Network Access; otherwise, I would've accepted 🤔
> > 2. Yes, so as soon I check the checkbox for the device, it goes into loading (or connecting mode), and the checkbox goes into the loading mode, but after a couple of seconds it just shows the error message above.
> > 3. Sure thing, I actually managed to connect to it with Razer Chroma Studio, so I know it's not the device. Absolutely, I'll give it a shot tonight and attach the logs and everything for you with the alpha version, if I didn't work.
> 
> @mamsoudi did you give a shot to the 2.5 version?

Sorry, my bad, absolutely forgot. I'm on it now.

**Comment by mamsoudi** [comment]
Right, so I installed the version `2.5.0`, it still shows the same error, I'm attaching the log file here if you wanna check it. Anything before **23 Jan** is with `2.4.0`. I can see it's throwing Timeout errors if it helps, I wasn't prompted for Network Access after installing `2.5.0` as well. 🤔

[RGBMaster.txt](https://github.com/rgb-master-team/RGBMaster/files/5860353/RGBMaster.txt)

**Comment by deankevorkian** [comment]
Thanks for the logs. Unfortunately they don't say much rather than the fact that connecting to the lights resulted in a timeout (just like you said). It still doesn't help me understand why, it simply means you've waited for 5 seconds and eventually gave up attempting to connect (usually it should take a few milliseconds).

Can you maybe add the app to your firewall rules manually? I still find it super strange that so many of our users aren't prompted with those messages. It feels like there's a more core issue that we're missing.

Other than that - we've had issues where people who used Yeelight Toolbox or other 3rd party apps, and then closed the app and opened RGBMaster, couldn't connect to their lights. Have you used any other 3rd party Yeelight apps before launching RGBMaster (since last reboot)? If you did, can you try to restart your pc and your lights (to reset all existing connections it has to other apps) and then try RGBMaster immediately after boot?

Thanks for helping!

**Comment by mamsoudi** [comment]
> Thanks for the logs. Unfortunately they don't say much rather than the fact that connecting to the lights resulted in a timeout (just like you said). It still doesn't help me understand why, it simply means you've waited for 5 seconds and eventually gave up attempting to connect (usually it should take a few milliseconds).
> 
> Can you maybe add the app to your firewall rules manually? I still find it super strange that so many of our users aren't prompted with those messages. It feels like there's a more core issue that we're missing.
> 
> Other than that - we've had issues where people who used Yeelight Toolbox or other 3rd party apps, and then closed the app and opened RGBMaster, couldn't connect to their lights. Have you used any other 3rd party Yeelight apps before launching RGBMaster (since last reboot)? If you did, can you try to restart your pc and your lights (to reset all existing connections it has to other apps) and then try RGBMaster immediately after boot?
> 
> Thanks for helping!

Yes I understand, I actually suspected that there might be something wrong with the firewall rules.

Definitely, I'll give it a shot tomorrow and let you know. I can perhaps try out both the stable and the alpha version. :)

**Comment by milovincent** [comment]
> I'll give it a shot tomorrow
...how did that go, i'm having the same issue
